### ISSUE TITLE
Smart Re-optimize showing the same URL for every skipped image

### ISSUE THREAD

**Issue by galbaras** [nosilver4u/ewww-image-optimizer#161]
**Describe the bug**
The bulk optimizer showed 330 images ready to be optimised. It then optimised 1 image, followed by 999 identical occurrences of another image, concluding with "Optimised 1000/330".

The second time around, there were only 999/329 images, but the same thing happened.

The "image" listed over and over was a PDF file.

**To Reproduce**
1. Go to Media > Bulk Optimise
2. Check the "Smart Re-optimize" box
3. Run the optimisation

**Expected behavior**
This is a new feature, so I'm not sure, but the media library shows an orange icon next to many images, so I'm guessing they should be re-optimised.

**Additional context**
Using IRSC for thumbnail regeneration, but I don't think this matters here.

**Comment by nosilver4u** [comment]
Seems the PDF record isn't getting removed from the queue somehow. Can you do the following:
1. Enable debugging on the support tab.
2. Click the link to remove the debug log (so we have a fresh start).
3. Run the scan with smart re-optimize active, and then start the optimization.
4. Let it run until it has repeated that PDF 3-4 times, and then hit the stop button.
5. Download the debug.log file from wp-content/ewww/ and send it to our support email.

Thanks1

**Comment by galbaras** [comment]
I had no chance of stopping. It got to 999 in a flash.

```
ewww_image_optimizer_settings_script()
ewww_image_optimizer_bulk_script()
not resuming/scanning, so clearing any pending images in both tables
load em all up
loading attachments into queue table
ewww_image_optimizer_insert_unscanned()
ewww_image_optimizer_db_init()
we have records to store via ewwwdb
ewww_image_optimizer_tool_script()
ewww_image_optimizer_webp_script()
ewww_image_optimizer_bulk_preview()
behappyinlife
EWWW IO version: 520.0
WP version: 5.3.2
PHP version: 70314
libxml version: 20904
core plugin
ewww_image_optimizer_cloud_quota()
quota data retrieved: {"licensed":"0","consumed":"53","days":"0","version":"3","metered":"1","recharge":0,"excredits":0,"amount_due":0.99}
ewww_image_optimizer_count_optimized()
scanning for media
ewww_image_optimizer_get_image_sizes()
sizes: thumbnail
medium
medium_large
large
1536x1536
2048x2048
half
featured
easy_testimonial_thumb
woocommerce_thumbnail
woocommerce_single
woocommerce_gallery_thumbnail
Array
(
[thumbnail] => Array
(
[width] => 150
[height] => 150
)
[medium] => Array
(
[width] => 554
[height] => 0
)
[medium_large] => Array
(
[width] => 768
[height] => 9999
)
[large] => Array
(
[width] => 750
[height] => 0
)
[1536x1536] => Array
(
[width] => 1536
[height] => 1536
)
[2048x2048] => Array
(
[width] => 2048
[height] => 2048
)
[half] => Array
(
[width] => 640
[height] => 640
)
[featured] => Array
(
[width] => 750
[height] => 750
)
[easy_testimonial_thumb] => Array
(
[width] => 50
[height] => 50
)
[woocommerce_thumbnail] => Array
(
[width] => 300
[height] => 0
)
[woocommerce_single] => Array
(
[width] => 360
[height] => 0
)
[woocommerce_gallery_thumbnail] => Array
(
[width] => 100
[height] => 100
)
[pdf-full] => Array
(
[width] => 99999
[height] => 99999
)
)

ewww_image_optimizer_aux_images()
ewww_image_optimizer_options()
behappyinlife
EWWW IO version: 520.0
WP version: 5.3.2
PHP version: 70314
libxml version: 20904
core plugin
ABSPATH: /home/behappyinlife/public_html/
WP_CONTENT_DIR: /home/behappyinlife/public_html/wp-content
home url: https://www.behappyinlife.com
site url: https://www.behappyinlife.com
content_url: https://www.behappyinlife.com/wp-content
upload_dir: /home/behappyinlife/public_html/wp-content/uploads
content_width: 
registered stream wrappers: https,ftps,compress.zlib,compress.bzip2,php,file,glob,data,http,ftp,phar,zip
ewww_image_optimizer_disable_tools()
ewww_image_optimizer_cloud_verify()
verification success via: https://optimize.exactlywww.com/verify/
verification body contents: {"status":"great"}
ewww_image_optimizer_cloud_quota()
quota data retrieved: {"licensed":"0","consumed":"53","days":"0","version":"3","metered":"1","recharge":0,"excredits":0,"amount_due":0.99}
ewww_image_optimizer_jpg_quality()
ewww_image_optimizer_skip_tools()
ewww_image_optimizer_gd_support()
GD found, supports:
GD Version: bundled (2.1.0 compatible)
FreeType Support: 1
FreeType Linkage: with freetype
GIF Read Support: 1
GIF Create Support: 1
JPEG Support: 1
PNG Support: 1
WBMP Support: 1
XPM Support: 1
XBM Support: 1
WebP Support: 1
BMP Support: 1
JIS-mapped Japanese Font Support: 
ewww_image_optimizer_gmagick_support()
ewww_image_optimizer_imagick_support()
3FR,AAI,AI,ART,ARW,AVI,AVS,BGR,BGRA,BGRO,BIE,BMP,BMP2,BMP3,BRF,CAL,CALS,CANVAS,CAPTION,CIN,CIP,CLIP,CMYK,CMYKA,CR2,CRW,CUR,CUT,DATA,DCM,DCR,DCX,DDS,DFONT,DNG,DOT,DPX,DXT1,DXT5,EPDF,EPI,EPS,EPS2,EPS3,EPSF,EPSI,EPT,EPT2,EPT3,ERF,EXR,FAX,FILE,FITS,FRACTAL,FTP,FTS,G3,G4,GIF,GIF87,GRADIENT,GRAY,GROUP4,GV,H,HALD,HDR,HISTOGRAM,HRZ,HTM,HTML,HTTP,HTTPS,ICB,ICO,ICON,IIQ,INFO,INLINE,IPL,ISOBRL,ISOBRL6,JBG,JBIG,JNG,JNX,JPE,JPEG,JPG,JPS,JSON,K25,KDC,LABEL,M2V,M4V,MAC,MAGICK,MAP,MASK,MAT,MATTE,MEF,MIFF,MKV,MNG,MONO,MOV,MP4,MPC,MPEG,MPG,MRW,MSL,MSVG,MTV,MVG,NEF,NRW,NULL,ORF,OTB,OTF,PAL,PALM,PAM,PANGO,PATTERN,PBM,PCD,PCDS,PCL,PCT,PCX,PDB,PDF,PDFA,PEF,PES,PFA,PFB,PFM,PGM,PICON,PICT,PIX,PJPEG,PLASMA,PNG,PNG00,PNG24,PNG32,PNG48,PNG64,PNG8,PNM,PPM,PREVIEW,PS,PS2,PS3,PSB,PSD,PTIF,PWP,RADIAL-GRADIENT,RAF,RAS,RAW,RGB,RGBA,RGBO,RGF,RLA,RLE,RMF,RW2,SCR,SCT,SFW,SGI,SHTML,SIX,SIXEL,SPARSE-COLOR,SR2,SRF,STEGANO,SUN,SVG,SVGZ,TEXT,TGA,THUMBNAIL,TIFF,TIFF64,TILE,TIM,TTC,TTF,TXT,UBRL,UBRL6,UIL,UYVY,VDA,VICAR,VID,VIFF,VIPS,VST,WBMP,WEBP,WMF,WMV,WMZ,WPG,X,X3F,XBM,XC,XCF,XPM,XPS,XV,XWD,YCbCr,YCbCrA,YUV
ewww_image_optimizer_savings()
querying savings for single site
table name is wpbhil_ewwwio_images
total original size: 231585911
total current(opt) size: 180875439
savings found: 50710472
5106 images have been optimized
automatic compression enabled
remove metadata: on
jpg level: 30
png level: 20
gif level: 10
pdf level: 10
bulk delay: 0
backup mode: on
ExactDN enabled: off
ExactDN all the things: on
ExactDN lossy: 1
ExactDN resize existing: off
ExactDN attachment queries: on
Easy IO exclusions:

lazy load: on
LQIP: off
LL exclusions:

ewww_image_optimizer_jpg_quality()
ewww_image_optimizer_jpg_quality()
effective quality: 82
parallel optimization: on
background optimization: on
scheduled optimization: off
include media library: off
include originals: off
folders to optimize:

folders to ignore:

skip images smaller than: 0 bytes
skip PNG images larger than: 0 bytes
exclude originals from lossy: off
exclude originals from metadata removal: off
use system binaries: no
resize detection: off
max media dimensions: 960 x 0
max other dimensions: 0 x 0
resize existing images: on
resize existing (non-media) images: off
ewww_image_optimizer_get_image_sizes()
sizes: thumbnail
medium
medium_large
large
1536x1536
2048x2048
half
featured
easy_testimonial_thumb
woocommerce_thumbnail
woocommerce_single
woocommerce_gallery_thumbnail
Array
(
[thumbnail] => Array
(
[width] => 150
[height] => 150
)
[medium] => Array
(
[width] => 554
[height] => 0
)
[medium_large] => Array
(
[width] => 768
[height] => 9999
)
[large] => Array
(
[width] => 750
[height] => 0
)
[1536x1536] => Array
(
[width] => 1536
[height] => 1536
)
[2048x2048] => Array
(
[width] => 2048
[height] => 2048
)
[half] => Array
(
[width] => 640
[height] => 640
)
[featured] => Array
(
[width] => 750
[height] => 750
)
[easy_testimonial_thumb] => Array
(
[width] => 50
[height] => 50
)
[woocommerce_thumbnail] => Array
(
[width] => 300
[height] => 0
)
[woocommerce_single] => Array
(
[width] => 360
[height] => 0
)
[woocommerce_gallery_thumbnail] => Array
(
[width] => 100
[height] => 100
)
[pdf-full] => Array
(
[width] => 99999
[height] => 99999
)
)

disabled resizes:
thumbnail: optimization=disabled creation=enabled
medium: optimization=enabled creation=enabled
medium_large: optimization=disabled creation=disabled
large: optimization=enabled creation=enabled
1536x1536: optimization=disabled creation=disabled
2048x2048: optimization=disabled creation=disabled
half: optimization=disabled creation=disabled
featured: optimization=disabled creation=disabled
easy_testimonial_thumb: optimization=disabled creation=disabled
woocommerce_thumbnail: optimization=enabled creation=enabled
woocommerce_single: optimization=enabled creation=enabled
woocommerce_gallery_thumbnail: optimization=disabled creation=enabled
pdf-full: optimization=enabled creation=enabled
delete originals: off
jpg2png: off
ewww_image_optimizer_jpg_background()
ewww_image_optimizer_jpg_background()
png2jpg: off
gif2png: on
webp conversion: on
alt webp rewriting: off
webp paths:

forced webp: off
enable help beacon: yes
ewww_image_optimizer_test_webp_mime_verify()
webp (real-world) verification succeeded
webp .htaccess rewriting enabled
max_execution_time: 300
ewww_image_optimizer_stl_check()
disable_functions: show_source, system, shell_exec, passthru, popen, proc_open, allow_url_fopen
detected memory limit is: 268435456
```

**Comment by nosilver4u** [comment]
I had the wrong mime-type (image/pdf vs. application/pdf) when checking the PDF compression level, thus we were getting a 0, and ALL PDFs were being flagged by the bulk optimizer.
But since this only affected the bulk scanner, the ewww_image_optimizer() function would look at the PDFs and ignore them.
Short version, it should work with the latest changes I checked in. Fixed that initial mime-type bug, and then made sure that a couple other edge cases couldn't cause problems.
